main.cpp

```diff
@@ -1466,15 +1489,16 @@ int main(int argc, char* argv[])
 		, std::ref(signals), std::ref(stats_timer), std::ref(ios)));
 
 	ios.run(ec);
+
+	for (auto& i : threads)
```

Comment:
the threads must be joined before destructed. the return below would cause a std::terminate in the thread destructors